src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/FilterMeta.cs

```diff
@@ -1,12 +1,53 @@
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
 namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
 
-public class FilterMeta
+public class FilterMeta : ICreatedUpdatedTimestamps<DateTimeOffset, DateTimeOffset?>
 {
-    public required string Identifier { get; set; }
+    public int Id { get; set; }
+
+    public required Guid DataSetVersionId { get; set; }
+
+    public DataSetVersion DataSetVersion { get; set; } = null!;
+
+    public required string PublicId { get; set; }
 
     public required string Label { get; set; }
 
     public string Hint { get; set; } = string.Empty;
 
-    public required List<FilterOptionMeta> Options { get; set; } = [];
+    public List<FilterOptionMeta> Options { get; set; } = [];
+
+    public List<FilterOptionMetaLink> OptionLinks { get; set; } = [];
+
+    public DateTimeOffset Created { get; set; }
+
+    public DateTimeOffset? Updated { get; set; }
+
+    internal class Config : IEntityTypeConfiguration<FilterMeta>
+    {
+        public void Configure(EntityTypeBuilder<FilterMeta> builder)
+        {
+            builder.Property(m => m.PublicId)
+                .HasMaxLength(100);
+
+            builder.HasIndex(m => new { m.DataSetVersionId, m.PublicId })
```

Comment:
Coming back to this...

I think I've figured out that `PublicIds` are unique when it comes to the options, but are not unique when it comes to the base filter. **(Please confirm/deny)**

But leaving this thread here just in case.

Reply:
**Discussed:**

`PublicId` is NOT UNIQUE across filters/versions for the `FilterMeta`, nor is it unique for the options either.